### STAT 207 Lab 5: Sampling Distribution, Standard Error and Confidence Interval

Tuan Tran-atran49

### SOLUTIONS

This lab consists of 2 main parts. You may add as many cells as you need in order to do the exercises. To add cells use the "Insert" tab from the menu bar above and select "insert cell above" or "insert cell below", and then select "code" or "Markdown" for the type of cell you want. 

**Note: not every problem in this lab requires Python calculation. In some cases you can just work it out and type in your solution.** 


### Part 1: Continuous Uniform Distribution (14 pts)

A random variable $X$ has a **uniform** distribution on the interval from $a$ to $b$ ($a < b$) if its probability density function (pdf) has a constant height $c=1/(b-a)$ over the interval from $a$ to $b$, and is zero outside the interval. The following questions in Part 1 refer to uniform random variables.

The model is available in SciPy as **scipy.stats.uniform** with the parameters **loc=a** (lower endpoint of interval) and **scale=b-a** (width of interval).


**a) (2 pts) Consider the uniform distribution with $a=-1$ and $b=4$. This is the distribution of a random real number between -1 and 4. Use scipy.stats.uniform to compute the height of the pdf between -1 and 4.**

In [1]:
import numpy as np
import pandas as pd
from scipy.stats import uniform
from scipy.stats import norm

In [2]:
uniform.pdf(0,loc=-1,scale=5)

0.2

**b) (2 pts) Explain how you could determine the height $c$ directly from the description of the uniform pdf, without using python. Recall that the total area under the pdf curve must equal 1.**

For uniform distribution on the interval from a to b (a<b). Height can be calculated:
$$
Area=Base\cdot Height= (b-a)\cdot c=1\\
\\c= \frac{1}{b-a}
$$

$$
C=\frac{1}{4-(-1)}=\frac{1}{5}=0.2
$$

**c) (2 pts) If X is a continuous random variable with a uniform distribution of [-1,4], use Python to calculate the mean $\sigma = E[\bar{x}]$ and the population standard deviation $\sigma$ for X.**

In [3]:
print("Mean: ",uniform.mean(loc=-1,scale=5))
print("SD: ",uniform.std(loc=-1,scale=5))

Mean:  1.5
SD:  1.4433756729740643


**d) (2 pts) Suppose $X_1, X_2, \ldots, X_{36}$ is a random sample of 36 observations from the uniform distribution on $[-1,4]$. Let $\bar{X} = {1\over 36} \sum_{i=1}^{36} X_i$. Compute the population mean, $E(\bar{X})$, and population standard deviation for $\bar{X}$. Hint: study the square root rule in the notes.**

In [4]:
pop_mean= uniform.mean(loc=-1,scale=5)
pop_std=(uniform.std(loc=-1,scale=5)/np.sqrt(36))
print("population mean: ",pop_mean)
print("population standard deviation: ", pop_std)

population mean:  1.5
population standard deviation:  0.24056261216234406


$$
E[\bar X]=\mu=1.5
$$

$$
\sigma=\sqrt{\frac{(b-a)^2}{12}}=\sqrt{\frac{5^2}{12}}=\sqrt{\frac{25}{12}} \approx 1.443
$$

$$
SD[\bar{X}] = {\sigma\over \sqrt{n}}= \frac{1.443}{\sqrt{36}}\approx 0.241
$$

**e) (4 pts) Use Python to generate a random sample of 36 observations from the uniform distribution on $[-1,4]$. Save them as a data frame or series and display them. In addition, compute and display the sample mean, the sample standard deviation, and the standard error for the sample mean $\bar{X}$. Show your work in the cells below.**

**i) Random sample:**

In [5]:
df=uniform.rvs(loc=-1,scale=5,size=36)
df

array([ 1.03892502e+00,  3.96457073e+00,  1.62267254e-02, -3.90239963e-01,
        8.67995469e-01,  4.89953172e-02,  1.20534562e+00,  1.20995511e+00,
        1.16001130e+00, -1.98422465e-01,  3.21251484e-01,  1.56962269e+00,
       -4.02005543e-01,  7.37262844e-02,  3.25399433e+00,  4.94929316e-01,
        2.31244135e+00,  2.90944766e+00,  7.98232889e-01,  1.37845773e+00,
        1.45002032e+00,  3.83114624e+00,  6.42385267e-01,  4.12905525e-01,
        9.96352269e-01,  2.17060897e+00,  3.48342804e+00, -5.72701914e-01,
        1.39452258e+00,  1.38019931e+00, -9.44639539e-01, -2.10777677e-03,
        3.02659315e+00,  4.70215531e-01,  5.48606274e-01,  4.51923537e-01])

**ii) Sample mean:**

In [6]:
print("mean: ",df.mean())

mean:  1.1214699681150837


**iii) Sample standard deviation:**

In [7]:
print("standard deviation: ",df.std())

standard deviation:  1.2561608929107893


**iv) Standard error for the sample mean:**

In [8]:
se=df.std()/np.sqrt(36)
print("standard error: ",se)

standard error:  0.20936014881846488


**f) (2 pts) According to the Central Limit Theorem, $\bar{X}$ is approximately normally distributed. You computed its population mean and standard deviation above. Use these results and the scipy 'norm' function to compute the approximate value for $P(\bar{X} \ge 2)$.**

In [9]:
1-norm.cdf(2,loc=pop_mean,scale=pop_std)

0.018833461114314387

### Part 2: Normal random sample (11 pts)

**Run the code below to generate a random sample from the normal distribution.**

In [10]:
import pandas as pd
import numpy as np
from scipy.stats import norm

In [11]:
samp = pd.Series(norm.rvs(loc=12.87, scale=3.14, size=20, random_state=1000))

**a) (2 pts) Compute and display the sample mean and sample standard deviation for the random sample.**

In [12]:
mean=samp.mean()
std=samp.std()
print("Sample mean",round(mean,3))
print("Sample standard deviation", round(std,3))

Sample mean 12.681
Sample standard deviation 2.036


**b) (1 pt) Compute the standard error for the sample mean, based on the generated data.**

In [13]:
n=20
se=round((std/np.sqrt(n)),3)
print("Standar error:",se)

Standar error: 0.455


**c) (2 pts) In order to compute an approximate 92% confidence interval for the population mean based on the sample, we need to compute the standard error multiplier, $z_{0.96}$. Compute $z_{0.96}$ using appropropriate python fuctions.**

In [14]:
zq=norm.ppf(0.96)
zq

1.7506860712521692

**d) (2 pts) calculate and display the endpoints of the approximate 92% confidence interval for the population mean $\mu$ based on the generated data.**

In [15]:
print("Confidence limits: ",(round(mean-zq*se,3),round(mean+zq*se,3)))

Confidence limits:  (11.885, 13.478)


**e) (1 pt) Does the interval you computed in d) contain the true population mean in this case? What is the numerical value of the population mean?**

Yes, the interval (11.876,13.487) contains the population mean $\bar x=12.87$

**f) (3 pts) If $\bar{X}$ is the sample mean for a random sample of size 20 generated using the method used here in Part 2 of the lab, what is the population distribution of $\bar{X}$? State the family and determine the numerical values for its population parameters such as mean and standard deviation.**

It is normal distribution. The mean of $\bar{X}$ will be the same with the mean of population ($E[\bar{X}]=\mu=12.87$). The standard deviation will be computed by applying square root theorem: $SD[\bar{X}] = {\sigma\over \sqrt{n}}={3.14 \over \sqrt{20}} \approx 0.7$